### ISSUE TITLE
[bug]: Dialog / AlertDialog not styled properly

### ISSUE THREAD

**Issue by bennaaym** [shadcn-ui/ui#5357]
### Describe the bug

**Description:**
I am encountering an issue with the styling of `Dialog` and `AlertDialog` components in my project. Despite following the documentation, the components do not render as expected, and the styling appears to be broken or missing. I am unsure of the root cause and have tried to troubleshoot without success.

```js
import { Button } from "@/lib/shadcn/ui/button";
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from "@/lib/shadcn/ui/dialog";

export const AlertDialog = () => {
  return (
    <Dialog>
      <DialogTrigger asChild>
        <Button variant="outline">Edit Profile</Button>
      </DialogTrigger>
      <DialogContent className="sm:max-w-[425px]">
        <DialogHeader>
          <DialogTitle>Edit profile</DialogTitle>
          <DialogDescription>
            Make changes to your profile here. Click save when you're done.
          </DialogDescription>
        </DialogHeader>
        <DialogFooter>
          <Button type="submit">Save changes</Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  );
};

```

![image](https://github.com/user-attachments/assets/7d298fc5-ce90-4666-9cef-657f16e07870)

**Expected Behavior:**
`Dialog` and `AlertDialog` should render with the proper styling as documented.

**Actual Behavior:**
The styling for `Dialog` and `AlertDialog` is either missing or broken. The components render with incorrect or no styles applied.

### Affected component/components

Dialog, AlertDialog

### How to reproduce

**Steps to Reproduce:**
1. Install ShadCN components as per the documentation.
2. Import and use `Dialog` or `AlertDialog` in a project.
3. Observe that the components do not have the correct or expected styling.

### Codesandbox/StackBlitz link

https://ui.shadcn.com/docs/components/dialog

### System Info

```bash
**Environment:**
- **Framework:** Next.js@14.2.15
- **Browser:** Chrome / Brave
- **OS:** Windows
- **Node.js version:** 20.14.0
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by Gurvinder-Singh02** [comment]
can you assign me this issue would love to resolve it

**Comment by bennaaym** [comment]
@Gurvinder-Singh02 Do you have an idea about what is causing such behavior?

**Comment by danielvanc** [comment]
Am hoping to use this Component asap, so would be good to know what's causing this bug.

**Comment by Gurvinder-Singh02** [comment]
Yeah i think it would be issue in tailwind.config but will figure out the root cause if you assign it to me

**Comment by bennaaym** [comment]
@imMatheus Tailwind is working as expected, all other shadcn component works as expected except for the Dialog and the AlertDialog

**Comment by imMatheus** [comment]
> @imMatheus Tailwind is working as expected, all other shadcn component works as expected except for the Dialog and the AlertDialog

@bennaaym Alright, could it be possible for you to create a codesandbox or similar, I cant seem to reproduce this issue :/

**Comment by imMatheus** [comment]
My guess would be as `Dialog` and `AlertDialog` are rendered in a portal, that could mess something up. Im thinking ig all other shadcn components are working but these components in a protal don't work, that seems like it could be the issue in some way :)

**Comment by IntrepidArrow** [comment]
As @imMatheus , the styling issues are occurring because these components are being rendered in a portal. 

I was experiencing some styling issues as well and I believe I was able to find a solution. I am using pages router, but I believe this should work for app router projects as well.

I was mainly having issues with typography in dialog and alert components. The font for these components was defaulting to times new roman, instead of jakarta sans which I chose for my project.

 1 - export my font variable from the `_app.tsx` file:
```
export const jakartaSans = Plus_Jakarta_Sans({
  subsets: ['latin'],
  variable: '--font-jakarta-sans',
});
```

2 - import this variable in the `_document.tsx` file and add the following to the `html` element:
```
import { Html, Head, Main, NextScript } from 'next/document';
import { jakartaSans } from './_app';
export default function Document() {
  return (
    <Html lang='en' className={jakartaSans.variable}>
      <Head />
      <body className='antialiased'>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}
```
Adding the font variable to the HTML element ensures that the font is available at the root level, which means it should be inherited by all elements, including portal elements.

I came up with this solution after reading through [Radix's typography documentation](https://www.radix-ui.com/themes/docs/theme/typography#custom-fonts), since I tried implementing my own dialog component usin Radix and experienced the same styling issues. 

If you are working with next.js's app router, it looks like Radix has some notes about common styling issues that you may encounter due to css import order when working with the app router: https://www.radix-ui.com/themes/docs/overview/styling#nextjs-import-order

If you are creating your own dialog or portal components using Radix, then you need to wrap your portal content with the `Theme` component. You can find this information in Radix's documentation here: https://www.radix-ui.com/themes/docs/overview/styling#missing-styles-in-portals

I hope this resolves your styling issues.

**Comment by girisht1** [comment]
Seriously, why can't we set a 'className' attribute on the 'AlertDialog' component ?

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to one year of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you. (This is an automated message)

**Comment by shadcn** [comment]
This issue has been automatically closed due to one year of inactivity. If you’re still experiencing a similar problem or have additional details to share, please open a new issue following our current issue template. Your updated report helps us investigate and address concerns more efficiently. Thank you for your understanding! (This is an automated message)
